base/abstractarraymath.jl

```diff
@@ -411,12 +411,26 @@ end
     eachrow(A::AbstractVecOrMat)
 
 Create a generator that iterates over the first dimension of vector or matrix `A`,
-returning the rows as views.
+returning the row vectors as views.
```

Comment:
I appreciate what you're trying to clarify, but I think this is confusing. Each row of the matrix gets returned, but as a vector, precisely not as a "row vector". 

Perhaps it's clearer just to say `view(r, :)` explicitly, something like
```suggestion
returning the rows as views, each a one-dimensional `view(r, :)` etc.
```

Reply:
I appreciate the further attempt at detail, but I don't think there is any room for confusion here, since "row vector" is not a type (there is no such thing in Julia), it is a vector as a row.

In any case, I included an example in both docstrings so everything should be clear to the reader and there should be absolutely no possibility of confusion. The primary focus of this PR is the examples.

Reply:
There is no RowVector, sure, but I would say the term covers `Transpose{..., Vector}` and friends, 1 x N objects. And I thought the goal was to clarify that such objects never appear here, they play no role. 

"The rows" of a matrix are pre-existing things, but there is no pre-existing collection of "the row vectors" unless you were to make one, which this function does not.

Reply:
> but there is no pre-existing collection of "the row vectors"

The concept of [row/column vectors](https://en.wikipedia.org/wiki/Row_and_column_spaces) is generally used in linear algebra, and is the same as here.

Reply:
Hmm... I guess we use to have `RowVector` which wasn’t an `AbstractVevtor`... I can see it might get confusing on occasionally. 

Maybe “... returning the rows as `AbstractVector` views”.

Reply:
Thanks, that should be clear, I made that change.